internal/envoy/v3/listener.go

```diff
@@ -486,6 +486,12 @@ func (b *httpConnectionManagerBuilder) Get() *envoy_listener_v3.Filter {
 			AcceptHttp_10:      true,
 			AllowChunkedLength: b.allowChunkedLength,
 		},
+		Http2ProtocolOptions: &envoy_core_v3.Http2ProtocolOptions{
+			MaxConcurrentStreams:        wrapperspb.UInt32(100),
```

Comment:
we may want to make this configurable on the Listener since the limit we have now is the Envoy one, which is the max allowed, users may be relying on this and we don't want to disturb working/valid traffic in an upgrade

Reply:
pulled this into https://github.com/projectcontour/contour/issues/5846